tokio-stream/src/stream_map.rs

```diff
@@ -568,6 +568,43 @@ where
     }
 }
 
+impl<K, V> std::iter::FromIterator<(K, V)> for StreamMap<K, V>
+where
+    K: Hash + Eq,
+{
+    fn from_iter<T: IntoIterator<Item = (K, V)>>(iter: T) -> Self {
+        let iterator = iter.into_iter();
+        let (lower_bound, _) = iterator.size_hint();
+        let mut stream_map = Self::with_capacity(lower_bound);
+
+        for (key, value) in iterator {
+            stream_map.insert(key, value);
+        }
+
+        stream_map
+    }
+}
+
+impl<K, V> IntoIterator for StreamMap<K, V> {
+    type Item = (K, V);
+    type IntoIter = std::vec::IntoIter<Self::Item>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.entries.into_iter()
+    }
+}
```

Comment:
Implementing `IntoIterator` for this type seems very confusing since it also implements `Stream`, but has a different item type as a `Stream`.

Reply:
You're right. I'm not sure if implementing `Extend` makes a lot of sense if `StreamMap` doesn't implement `IntoIterator`. I removed both Extend and IntoIterator impls if you feel like Extend makes sense I can add it back.